From 1b096582338c1957e610224627eee7cd7650c0a9 Mon Sep 17 00:00:00 2001
From: "Yang, Dong" <dong.yang@intel.com>
Date: Mon, 24 Aug 2015 15:56:32 +0800
Subject: [PATCH] fix the antibanding doesn't work issue

the antibanding function doesn't work due to the wrong config of
frame_length_lines and line_length_pck, set them to the correct
value, so the 3A module can calculate the exposure time correctly

Change-Id: I1f9484bfaa3bf56d457f8d6f73d886c86b20de87
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-13198
Signed-off-by: Yang, Dong <dong.yang@intel.com>
---
 .../camera/drivers/media/i2c/gc2235.c              |  6 ++-
 .../camera/drivers/media/i2c/gc2235.h              | 48 +++++++++++-----------
 2 files changed, 29 insertions(+), 25 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/gc2235.c b/drivers/external_drivers/camera/drivers/media/i2c/gc2235.c
index 32e1026..a079e06 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/gc2235.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/gc2235.c
@@ -267,7 +267,7 @@ static int gc2235_get_intg_factor(struct i2c_client *client,
 		return -EINVAL;
 
 	/* pixel clock calculattion */
-	buf->vt_pix_clk_freq_mhz = dev->vt_pix_clk_freq_mhz = 43750000;
+	buf->vt_pix_clk_freq_mhz = dev->vt_pix_clk_freq_mhz = 30000000;
 
 	/* get integration time */
 	buf->coarse_integration_time_min = GC2235_COARSE_INTG_TIME_MIN;
@@ -337,8 +337,10 @@ static int gc2235_get_intg_factor(struct i2c_client *client,
 	ret = gc2235_read_reg(client, GC2235_8BIT,
 					GC2235_SH_DELAY_L, &reg_val);
 
+#if 0
 	buf->line_length_pck = buf->output_width + 16 + dummy +
 				(((u16)reg_val_h << 8) | (u16)reg_val) + 4;
+#endif
 	ret = gc2235_read_reg(client, GC2235_8BIT,
 					GC2235_VB_H, &reg_val_h);
 	ret = gc2235_read_reg(client, GC2235_8BIT,
@@ -346,8 +348,10 @@ static int gc2235_get_intg_factor(struct i2c_client *client,
 	if (ret)
 		return ret;
 
+#if 0
 	buf->frame_length_lines = buf->output_height + 32 +
 				(((u16)reg_val_h << 8) | (u16)reg_val);
+#endif
 	buf->binning_factor_x = res->bin_factor_x ?
 					res->bin_factor_x : 1;
 	buf->binning_factor_y = res->bin_factor_y ?
diff --git a/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h b/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
index eb5b380..d64b32a 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/gc2235.h
@@ -543,11 +543,11 @@ struct gc2235_resolution gc2235_res_preview[] = {
 		.desc = "gc2235_1600_900_30fps",
 		.width = 1600,
 		.height = 900,
-		.pix_clk_freq = 60,
+		.pix_clk_freq = 30,
 		.fps = 30,
 		.used = 0,
-		.pixels_per_line = 1616,
-		.lines_per_frame = 932,
+		.pixels_per_line = 2132,
+		.lines_per_frame = 1068,
 		.bin_factor_x = 0,
 		.bin_factor_y = 0,
 		.bin_mode = 0,
@@ -559,11 +559,11 @@ struct gc2235_resolution gc2235_res_preview[] = {
 		.desc = "gc2235_1600_1066_30fps",
 		.width = 1616,
 		.height = 1082,
-		.pix_clk_freq = 60,
+		.pix_clk_freq = 30,
 		.fps = 30,
 		.used = 0,
-		.pixels_per_line = 1616,
-		.lines_per_frame = 1098,
+		.pixels_per_line = 2132,
+		.lines_per_frame = 1368,
 		.bin_factor_x = 0,
 		.bin_factor_y = 0,
 		.bin_mode = 0,
@@ -574,11 +574,11 @@ struct gc2235_resolution gc2235_res_preview[] = {
 		.desc = "gc2235_1600_1200_30fps",
 		.width = 1616,
 		.height = 1216,
-		.pix_clk_freq = 60,
+		.pix_clk_freq = 30,
 		.fps = 30,
 		.used = 0,
-		.pixels_per_line = 1616,
-		.lines_per_frame = 1232,
+		.pixels_per_line = 2132,
+		.lines_per_frame = 1368,
 		.bin_factor_x = 0,
 		.bin_factor_y = 0,
 		.bin_mode = 0,
@@ -594,11 +594,11 @@ struct gc2235_resolution gc2235_res_still[] = {
 		.desc = "gc2235_1600_900_30fps",
 		.width = 1600,
 		.height = 900,
-		.pix_clk_freq = 60,
+		.pix_clk_freq = 30,
 		.fps = 30,
 		.used = 0,
-		.pixels_per_line = 1616,
-		.lines_per_frame = 932,
+		.pixels_per_line = 2132,
+		.lines_per_frame = 1068,
 		.bin_factor_x = 0,
 		.bin_factor_y = 0,
 		.bin_mode = 0,
@@ -609,11 +609,11 @@ struct gc2235_resolution gc2235_res_still[] = {
 		.desc = "gc2235_1600_1066_30fps",
 		.width = 1616,
 		.height = 1082,
-		.pix_clk_freq = 60,
+		.pix_clk_freq = 30,
 		.fps = 30,
 		.used = 0,
-		.pixels_per_line = 1616,
-		.lines_per_frame = 1098,
+		.pixels_per_line = 2132,
+		.lines_per_frame = 1368,
 		.bin_factor_x = 0,
 		.bin_factor_y = 0,
 		.bin_mode = 0,
@@ -624,11 +624,11 @@ struct gc2235_resolution gc2235_res_still[] = {
 		.desc = "gc2235_1600_1200_30fps",
 		.width = 1616,
 		.height = 1216,
-		.pix_clk_freq = 60,
+		.pix_clk_freq = 30,
 		.fps = 30,
 		.used = 0,
-		.pixels_per_line = 1616,
-		.lines_per_frame = 1232,
+		.pixels_per_line = 2132,
+		.lines_per_frame = 1368,
 		.bin_factor_x = 0,
 		.bin_factor_y = 0,
 		.bin_mode = 0,
@@ -644,11 +644,11 @@ struct gc2235_resolution gc2235_res_video[] = {
 		.desc = "gc2235_1296_736_30fps",
 		.width = 1296,
 		.height = 736,
-		.pix_clk_freq = 60,
+		.pix_clk_freq = 30,
 		.fps = 30,
 		.used = 0,
-		.pixels_per_line = 1296,
-		.lines_per_frame = 736,
+		.pixels_per_line = 1828,
+		.lines_per_frame = 888,
 		.bin_factor_x = 0,
 		.bin_factor_y = 0,
 		.bin_mode = 0,
@@ -659,11 +659,11 @@ struct gc2235_resolution gc2235_res_video[] = {
 		.desc = "gc2235_960_640_30fps",
 		.width = 960,
 		.height = 640,
-		.pix_clk_freq = 60,
+		.pix_clk_freq = 30,
 		.fps = 30,
 		.used = 0,
-		.pixels_per_line = 976,
-		.lines_per_frame = 656,
+		.pixels_per_line = 1492,
+		.lines_per_frame = 792,
 		.bin_factor_x = 0,
 		.bin_factor_y = 0,
 		.bin_mode = 0,
-- 
1.9.1

